peer.go

```diff
@@ -1245,10 +1249,12 @@ func (p *peer) handlePongMsg(msg *btcwire.MsgPong) {
 	// infrequently enough that if they overlap we would have timed out
 	// the peer.
 	if p.protocolVersion > btcwire.BIP0031Version &&
-		p.lastPingNonce != 0 && msg.Nonce == p.lastPingNonce {
-		p.lastPingMicros = time.Now().Sub(p.lastPingTime).Nanoseconds()
-		p.lastPingMicros /= 1000 // convert to usec.
-		p.lastPingNonce = 0
+		msg.Nonce == p.lastPingNonce {
+		if p.pingTimer != nil {
```

Comment:
I'm fairly certain this is a race.  Both the `inHandler` and `outHandler` goroutines are accessing the `pingTimer` field on the peer without any synchronization to protect access.